src/rest/commands/general/follow.ts

```diff
@@ -0,0 +1,131 @@
+import Command, {
+  basicInteractionResponse,
+} from "../../common/command/Command.ts";
+import { bot } from "../../cache.ts";
+import { enTranslate, translate } from "../../common/util/i18next.ts";
+import {
+  DiscordApplicationCommandOptionTypes,
+  followChannel,
+  getChannelWebhooks,
+  snowflakeToBigint,
+  validatePermissions,
+} from "../../../../deps.ts";
+
+const FollowCommand: Command = async (interaction) => {
+  const guildId: bigint = snowflakeToBigint(interaction.guildId!);
+
+  if (
+    !validatePermissions(snowflakeToBigint(interaction.member!.permissions), [
+      "MANAGE_WEBHOOKS",
+    ])
+  ) {
+    return basicInteractionResponse(
+      interaction.id,
+      interaction.token,
+      translate(guildId, "permission:USER_MISSING_PERMISSION", {
+        permission: "Manage Webhooks",
+      }),
+    );
+  }
+
+  const raw = interaction.data?.options?.[0];
+
+  if (raw?.name === "announcements") {
+    const channelId: bigint = snowflakeToBigint(interaction.channelId!);
+
+    const channelWebhooks = await getChannelWebhooks(
+      channelId,
+    );
+
+    if (channelWebhooks.size >= 10) {
+      return basicInteractionResponse(
+        interaction.id,
+        interaction.token,
+        translate(guildId, "commands/general/follow:WEBHOOK_LIMIT"),
+      );
+    }
+
+    try {
+      // 268559149175013376 is the ID of our announcement channel.
+      await followChannel(snowflakeToBigint("268559149175013376"), channelId);
```

Comment:
Should this be an env option, in case this changes in the future for whatever reason?